### ISSUE TITLE
Should we run travis test suite for PR's labelled "WIP"

### ISSUE THREAD

**Issue by MasterOdin** [Submitty/Submitty#2409]
We've been using the [WIP] tag on PR's to designate that they're ready for preliminary code review but are going to still undergo some number of continued patches of work and that as is, they are probably not ready to be merged into master. However, each push to these branches kick off a push and PR build on Travis which ends up hogging the limited number of machines we have available and drastically lowering the throughput available.

Should we set up Travis such that if it detects a pull request and the title of the pull request is [WIP], then just auto-fail. People will still have the push build to look at for preliminary information about the build and whether it's passing tests or not.

Implementation would be that we add a new Travis Job stage that simply checks:
- if TRAVIS_EVENT_TYPE == "pull_request
- and if the title attribute of `curl https://api.github.com/repos/Submitty/Submitty/pulls/${TRAVIS_PULL_REQUEST}` starts with "[WIP]"
- then `exit 1`
- else `exit 0`

(example API call for a PR that has a WIP progress tag: https://api.github.com/repos/Submitty/Submitty/pulls/2321)

Once a PR is ready, remove the WIP tag on the title, and then either rerun the latest build or make a new commit (merge master or something being an easy candidate).

@bmcutler: thoughts?

**Comment by bmcutler** [comment]
Sure, this sounds like a good plan.

Also, I propose [Initial Work] as the tag that means 'this is part 1 of a multi-part thing, we can merge it into master now (it doesn't break things) but people skimming the commit history should know this feature isn't done yet'
